src/Stryker.Core/Stryker.Core/Mutators/BlockMutator.cs

```diff
@@ -13,7 +15,53 @@ class BlockMutator : MutatorBase<BlockSyntax>, IMutator
 
         public override IEnumerable<Mutation> ApplyMutations(BlockSyntax node)
         {
-            if (node.Parent is MethodDeclarationSyntax methodDeclaration && methodDeclaration.ReturnType.ToString() != "void")
+            // Note: using local functions below so that the check order can be easily reordered,
+            // if a need to optimize arises at some point.
+
+            bool IsInStructConstructor()
+            {
+                foreach (var ancestor in node.Ancestors())
+                {
+                    switch (ancestor)
+                    {
+                        // Don't count local functions as being part of the constructor.
+                        // They can't assign to members, so they can be mutated.
+                        case LocalFunctionStatementSyntax:
```

Comment:
This case is not needed right? Which kinda makes the whole switch unneeded. I think it could just be an if statement 👍

Reply:
`ShouldMutateLocalFunctionsInStructConstructors` breaks if this is removed. It needs to return early in in a local function within a struct constructor.

Reply:
Ah I get that now